### ISSUE TITLE
clarification of build procedure

### ISSUE THREAD

**Issue by jamie-arcc** [mphowardlab/azplugins#68]
I am having a really hard time getting azplugins to build and install...  I have tried both the main branch as well as the 0.11.0 release and I am trying to build against hoomd 2.9.7.  For the internal build method, is `cmake` supposed to be run after linking the azplugins directory in? if I run cmake, link in azplugins, and run `make install`, azplugins doesn't get installed, just hoomd.  if I link it in before running cmake, the cmakefiles for azplugins complain that they can't find hoomd.  if I try the external build method after installing hoomd, I get two undefined symbol references at the end.

build environment:
- CentOS 7 HPC Cluster
- CUDA 11.5
- gcc 9.1.0
- python 3.8.0

any assistance or clarification you can provide would be greatly appreciated

**Comment by jamie-arcc** [comment]
I just realized I was symlinking the top `azplugins` directory rather than `azplugins/azplugins`

**Comment by mphoward** [comment]
For an internal build, you should symlink then run cmake. Are you symlinking the right directory? You want to link `azplugins/azplugins` (the python module & source code) and not the base `azplugins`. I suspect you are getting the base directory, which would cause cmake to run `FindHOOMD.cmake` (and fail).

For an external build, I would need to see the error you are getting for undefined symbols. We run our unit tests as external builds (without error) in a similar build configuration.

See the HOOMD documentation here: https://hoomd-blue.readthedocs.io/en/v2.9.7/developer.html#plugins-and-components

**Comment by mphoward** [comment]
Is this error from `0.11.0` or `main`? I committed on `main` to fix a thrust/cub compile issue, but I don't think we tagged a release:

https://github.com/mphowardlab/azplugins/commit/3d9a7beac0645c9c1116ef0bf05965037a4b731e

Let me know if this fixes the problem for you, and I will release a version `0.12.0` since there are actually a couple commits there that are not in the changelog.

**Comment by mphoward** [comment]
I am noticing that this commit may not work for an internal build because of where the cmake code was added... please confirm the error is present on `main`, and I'll have a look.

**Comment by jamie-arcc** [comment]
I was using `0.11.0` originally.  It looks like it compiled fine when I switched to the `main` branch, though it did throw a bunch of warnings about namespaces being redefined:
```
<command-line>: warning: "THRUST_CUB_WRAPPED_NAMESPACE" redefined
<command-line>: note: this is the location of the previous definition
<command-line>: warning: "HOOMD_THRUST" redefined
<command-line>: note: this is the location of the previous definition
<command-line>: warning: "HOOMD_CUB" redefined
<command-line>: note: this is the location of the previous definition
```

but they're just warnings. thanks a ton for being so responsive and helpful!

**Comment by mphoward** [comment]
OK thanks! That's good to know, I will have to look to see how HOOMD is defining those themselves, as I know they are needed in each module. This bug fix is only sometimes needed for CUDA 11.5 and I think we removed in HOOMD v3, so I haven't worried too much about it.

**Comment by mphoward** [comment]
Cleaning house: the warning noted above was for a specific version of CUDA that we have moved a few versions past, so I am closing this issue.
